From f44ca5f2c9c464f0515b2f4404657a5987b47d46 Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Tue, 4 Dec 2012 16:51:17 +0100
Subject: [PATCH 2/2] Improve ipa-replica-prepare error message

When DNS zone/record manipulation commands fails for example due to
a ValidationError, ipa-replica-prepapre reports a whole traceback
which is difficult to read. Make sure our error error is more
readable.

https://fedorahosted.org/freeipa/ticket/3283
---
 install/tools/ipa-replica-prepare | 21 +++++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/install/tools/ipa-replica-prepare b/install/tools/ipa-replica-prepare
index d67ed818ec39e6d2800b3c0ab23a5370ee14fe3e..6d01c4c0017ab7983d3cea6dd6b1142dd8002fea 100755
--- a/install/tools/ipa-replica-prepare
+++ b/install/tools/ipa-replica-prepare
@@ -463,13 +463,26 @@ def main():
             if reverse_zone is None and not options.no_reverse:
                 reverse_zone = bindinstance.get_reverse_zone_default(ip)
 
-        add_zone(domain)
-        add_fwd_rr(domain, name, ip_address)
+        try:
+            add_zone(domain)
+        except errors.PublicError, e:
+            sys.exit("Could not create forward DNS zone for the replica: %s" % e)
+
+        try:
+            add_fwd_rr(domain, name, ip_address)
+        except errors.PublicError, e:
+            sys.exit("Could not add forward DNS record for the replica: %s" % e)
 
         if reverse_zone is not None:
             print "Using reverse zone %s" % reverse_zone
-            add_zone(reverse_zone)
-            add_ptr_rr(reverse_zone, ip_address, replica_fqdn)
+            try:
+                add_zone(reverse_zone)
+            except errors.PublicError, e:
+                sys.exit("Could not create reverse DNS zone for the replica: %s" % e)
+            try:
+                add_ptr_rr(reverse_zone, ip_address, replica_fqdn)
+            except errors.PublicError, e:
+                sys.exit("Could not add reverse DNS record for the replica: %s" % e)
 
 try:
     if not os.geteuid()==0:
-- 
1.7.11.7

